src/subtyping.md

```diff
@@ -15,5 +15,71 @@ fn bar<'a>() {
     let t: &'a str = s;
 }
 ```
-Since `'static` "lives longer" than `'a`, `&'static str` is a subtype of
-`&'a str`.
+
+Since `'static` outlives `'a`, `&'static str` is a subtype of `&'a str`.
+
+Subtyping also exists for [higher-ranked types]. Replacing a higher ranked
+lifetime with a concrete lifetime produces a subtype.
+
+```rust
+fn bar() {
+    // Explicitly f: for<'a> fn(&'a i32) -> &'a i32.
+    let f: fn(&i32) -> &i32 = |x| x;
+    let g: fn(&'static i32) -> &'static i32 = f;
+}
+```
+
+The subtype must be a valid type:
+
+```rust,compile_fail
+fn bar<'a>() {
+    let h: for<'b, 'c> fn(&'c &'b i32) -> &'b i32 = |x| &**x;
+    let j: fn(&'static &'a i32) -> &'a i32 = h;
+    // Error: in type `&'static &'a i32`, reference has a longer lifetime than
+    // the data it references
+}
+```
```

Comment:
Unclear to me why this merits discussion (is this meaningfully different from adding a note that "the expression must be parse"?)